src/pyfaf/celery_tasks/__init__.py

```diff
@@ -0,0 +1,75 @@
+from celery import Celery
+from celery.signals import task_postrun
+from pyfaf.storage import DatabaseFactory, TaskResult
+from pyfaf.actions import actions
+from pyfaf.config import config
+import bunch
+import sys
+import logging
+import datetime
+from cStringIO import StringIO
+
+celery_app = Celery("pyfaf_tasks",
+                    broker=config.get("celery_tasks.broker", ""),
+                    backend=config.get("celery_tasks.backend", ""))
+
+db_factory = DatabaseFactory(autocommit=True)
+
+
+class ActionError(Exception):
+    def __init__(self, output):
+        self.output = output
+        Exception.__init__(self, output)
+
+    def __str__(self):
+        return self.output
+
+
+@task_postrun.connect
+def task_postrun_handler(signal, sender, **named):
+    db = db_factory.get_database()
+    tr = TaskResult()
+    tr.id = named["task_id"]
+    tr.task = named["task"].name
+    tr.finished_time = datetime.datetime.now()
+    tr.state = named["state"]
+    tr.retval = named["retval"]
+    tr.args = named["args"]
+    tr.kwargs = named["kwargs"]
+    db.session.add(tr)
+    db.session.flush()
+
+
+@celery_app.task()
+def run_action(name, params={}, log_level=logging.DEBUG, output_length=1000):
+    db = db_factory.get_database()
+    action = actions[name]
+    cmdline = bunch.Bunch(params)
+
+    # Logger and fake stdout to capture the output of the action
+    my_stdout = StringIO()
```

Comment:
Please use capture_output context manager:

```
from pyfaf.utils.contextmanager import captured_output
with captured_output() as (cap_stdout, cap_stderr):
    <code>
```

cap_std* are StringIO objects, sys.stdout/err is saved/restored automatically.